crates/typst-pdf/src/page.rs

```diff
@@ -229,17 +229,25 @@ fn write_page(ctx: &mut PdfContext, i: usize, resources_ref: Ref) {
 
 /// Write the page labels.
 pub(crate) fn write_page_labels(ctx: &mut PdfContext) -> Vec<(NonZeroUsize, Ref)> {
+    // If there is no page labeled, we skip the writing
+    if !ctx.pages.iter().any(|p| {
+        p.label
+            .as_ref()
+            .is_some_and(|l| l.prefix.is_some() || l.style.is_some())
+    }) {
+        return Vec::new();
+    }
+
+    // If there pages with empty labels between labeled pages, we must write
+    // empty PageLabel entries.
```

Comment:
```suggestion
    // If there are pages with empty labels between labeled pages, we must write
    // empty PageLabel entries.
```
and should be moved below directly above the `page.label.as_ref().unwrap_or(&empty_label);` line

Reply:
I applied the suggestion on commit  3f983fd0a27d8ee7c3687ee9c27a6771c61353b2.